src/arch/ia32e/descriptor/gate.rs

```diff
@@ -1,92 +1,83 @@
-use crate::arch::ia32e::{
-    selector::SegmentSelector, PrivilegeLevel};
+use crate::arch::ia32e::{selector::SegmentSelector, PrivilegeLevel};
+use bit_field::BitField;
+use configuration::Configuration;
 use core::fmt;
-use crate::arch::ia32::descriptor::gate::lower::Lower;
-use upper::Upper;
 
-mod upper;
+mod configuration;
 
-#[repr(u8)]
+#[repr(u16)]
 pub enum Kind {
     Interrupt = 0xe,
     Trap = 0xf,
 }
 
+impl From<Kind> for u16 {
+    fn from(value: Kind) -> Self {
+        value as u16
+    }
+}
+
 #[must_use]
 #[derive(Default, Copy, Clone)]
 #[repr(C, packed)]
 pub struct Gate {
-    /// Segment selector bits (0:31).
-    lower: Lower,
-    ///  Segment selector bits (32:64).
-    upper: Upper,
-    /// Procedure's entry point offset high bits.
+    offset_15_0: u16,
```

Comment:
Don't you want to comment the role of every field as you did for other structs ?

Reply:
Fixed in fe6ea60